extensions/eclipse-che-theia-plugin-remote/src/node/hosted-plugin-remote.ts

```diff
@@ -177,4 +192,50 @@ export class HostedPluginRemote {
         return [].concat.apply([], [...this.pluginsMetadata.values()]);
     }
 
+    /**
+     * Sends request to retreive plugin resource from its sidecar.
+     * Returns undefined if plugin doesn't run in sidecar or doesn't exist.
+     * @param pluginId id of the plugin for which resource should be retreived
+     * @param resourcePath relative path of the requested resource based on plugin root directory
+     */
+    public requestPluginResource(pluginId: string, resourcePath: string): Promise<Buffer | undefined> {
+        if (this.hasEndpoint(pluginId) && resourcePath) {
```

Comment:
I think it will more readable
```
if (!this.hasEndpoint(pluginId) || !resourcePath) {
    return undefined;
}

return new Promise((resolve, reject) => {
 ...
});

```

Reply:
WDYT?

Reply:
Both variants have their own pros and cons. Your case is clearer in term of return value but has two negation in the expression. To me both are ok.